Silicon/TigerlakePkg/Library/TxtInit/TxtCtxLib.h

```diff
@@ -0,0 +1,300 @@
+  @param[in]  Funct        Function number of BIOS ACM to be executed
+
+  @retval     EFI_SUCCESS  Always.
+**/
+VOID
+LaunchBiosAcm (
+  IN UINT64 AcmBase,
+  IN UINT64 Funct
+  );
+
+/**
+  Sets up the system and then launches the TXT BIOS ACM to run the function
+  requested by AcmFunction.
+
+  @param[in]  AcmFunction    Constant that represents the function from the BIOS ACM
+                             that should be executed.
+  @param[in]  TxtDxeCtx      A pointer to an initialized TXT DXE Context data structure
+
+  @retval     EFI_SUCCESS    Always.
+**/
+EFI_STATUS
+TxtDxeLibLaunchBiosAcm (
+  IN TXT_DXE_LIB_CONTEXT *TxtDxeCtx,
+  IN UINT64              AcmFunction
+  );
+
+/**
+  AP initial routine executed through MP service for TXT SCHECK
+
+  @param[in]  Buffer  Not used.
+**/
+VOID
+EFIAPI
+DoApInit (
+  IN VOID *Buffer
+  );
+
+/**
+  Save AP configuration routine executed through MP service for TXT SCHECK
+
+  @param[in]  Buffer  Not used.
+**/
+VOID
+EFIAPI
+ApSaveConfig (
+  IN VOID *Buffer
+  );
+
+/**
+  Restore AP configuration routine executed through MP service for TXT SCHECK
+
+  @param[in]  Buffer  Not used.
+**/
+VOID
+EFIAPI
+ApRestoreConfig (
+  IN VOID *Buffer
+  );
+
+/**
+  Disable or restore possible SMI sources before or after POST SCHECK
+
+  @param[in]  TxtDxeCtx   A pointer to an initialized TXT DXE Context data structure
+  @param[in]  Operation   Boolean value telling what operation is requested:
+                          TRUE  - to save and then disable possible SMI sources
+                          FALSE - to restore original SMI settings
+
+  @retval    EFI_SUCCESS  always return EFI_SUCCESS
+**/
+EFI_STATUS
+DisableSmiSources (
+  IN TXT_DXE_LIB_CONTEXT *TxtDxeCtx,
+  IN BOOLEAN             Operation
+  );
+
+/**
+  This function gets called before/after run the SCHECK function, intend to avoid platform specific SMIs to interfere in BIOS POST
+  if BIOS SMM MP services or synchonization code is not well considered that some of APs are not waken up from Wait-for-SIPI state.
+  Function should preserve original SMI enabling setting in augument is TRUE, and then restore it in augurment is FALSE.
+  The caller is DisableSmiSources() in TxtDxeLib.c
+
+  @param[in]  TxtDxeCtx
+  @param[in]  Operation = TRUE     Calling before SCHECK to saved and disable platform specific SMIs setting
+                       = FALSE    Calling after SCHECK to restore platform specific SMIs setting
+
+  @retval     EFI_SUCCESS          Always.
+**/
+extern
+EFI_STATUS
+EFIAPI
+TxtDxeOemDisableSmi (
+  IN TXT_DXE_LIB_CONTEXT *TxtDxeCtx,
+  IN BOOLEAN             Operation
+  );
+
+/**
+  Read policy protocol to reset AUX content
+
+  @param[in]  TxtDxeCtx       A pointer to an initialized TXT DXE Context data structure
+
+  @retval     EFI_SUCCESS     No error happend
+  @retval     EFI_NOT_FOUND   TxtPolicyProtocol is not found
+**/
+EFI_STATUS
+ResetTpmAux (
+  IN TXT_DXE_LIB_CONTEXT *TxtDxeCtx
+  );
+
```

Comment:
There is no TxtCtxLib, maybe rename it to TxtCtx.h.
All the functions are for internal use, EFIAPI could be removed in all functions.

Reply:
done